plugins/single_plugins/wm-actions.cpp

```diff
@@ -40,20 +40,35 @@ class wayfire_wm_actions_t : public wf::plugin_interface_t
         return true;
     };
 
+    wf::activator_callback on_toggle_fullscreen =
+        [=](wf::activator_source_t source, uint32_t) -> bool
+    {
+        if (!output->can_activate_plugin(this->grab_interface))
+            return false;
+
+        auto view = choose_view(source);
```

Comment:
I believe this would need the same checks as the `on_toggle_above` code has, namely, whether the view's role is TOPLEVEL. We don't want to fullscreen panels or such stuff ...

Reply:
That makes sense, I have pushed an update, thanks for review. :slightly_smiling_face: